### PR TITLE
remove unnecessary call

### LINKED ISSUE #55158: MsQuic configuration handle is created but not used
See https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs#L175

### PR DESCRIPTION
**Pull request by wfurt**
We call SafeMsQuicConfigurationHandle.Create twice but used only once configuration. 
Since we do not need it past connect I Dispose it there so we don't drag it forward.

Fixes #55158

### CODE REVIEW

**Comment by ManickaP** [comment]
Does this closes #55158. Could you link it here and close it with this PR if so?

**Comment by wfurt** [comment]
fixes  #55158

**Comment by karelz** [comment]
@wfurt it needs to be in top post to work with GH -- I added it there ... just FYI

**Review by CarnaViire — APPROVED** [review]

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ internal sealed class MsQuicConnection : QuicConnectionProvider
 
         // TODO: remove this.
         // This is only used for client-initiated connections, and isn't needed even then once Connect() has been called.
-        private readonly SafeMsQuicConfigurationHandle? _configuration;
+        private SafeMsQuicConfigurationHandle? _configuration;
```
Comment (geoffkizer) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs:30]: Given the comment here, I wonder why we don't just create (and dispose) this in ConnectAsync so we don't have to store the value at all.

Comment (wfurt) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs:30]: I think we may. That would possibly change where exception can be thrown e.g. in constructor were the options are passed in or  in Connect. We would need to store all the options too, right?

Comment (geoffkizer) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs:30]: Yeah, I suppose we need to store the options as well.

Though alternatively, we could pass options to ConnectAsync and never have to store them at all. I don't think we should be making that kind of API change right now, but we can certainly consider it for 7.0.

Comment (geoffkizer) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs:30]: @wfurt I filed an issue for 7.0 to consider changing the API here to make things simpler, here: https://github.com/dotnet/runtime/issues/56009

Please take a look at the issue and make sure it looks good to you.

**Pull request merged**